src/librustc_mir/transform/check_consts/validation.rs

```diff
@@ -230,12 +230,11 @@ impl Validator<'a, 'mir, 'tcx> {
     }
 
     fn check_static(&mut self, def_id: DefId, span: Span) {
-        let is_thread_local = self.tcx.has_attr(def_id, sym::thread_local);
-        if is_thread_local {
-            self.check_op_spanned(ops::ThreadLocalAccess, span)
-        } else {
-            self.check_op_spanned(ops::StaticAccess, span)
-        }
+        assert!(
+            !self.tcx.has_attr(def_id, sym::thread_local),
```

Comment:
Maybe use `!self.tcx.codegen_fn_attrs(def_id).flags.contains(CodegenFnAttrFlags::THREAD_LOCAL)` instead?

Reply:
that's a mouthful :laughing: 

I just used what we used elsewhere, also at this point, codegen doesn't need to know about flags on statics anymore, so the notion of `CodegenFnAttrFlags::THREAD_LOCAL` seems wrong in two accounts:

1. it's for attributes on functions
2. it's not something codegen needs to know about on statics

Reply:
`cg_llvm` uses this in `get_static` for cross-crate statics.

> it's for attributes on functions

It also contains linkage, `link_name`, `no_mangle`, ..., which are important for statics.

https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/middle/codegen_fn_attrs/struct.CodegenFnAttrs.html

Reply:
oh, ok that's oddly named then. I guess we could use that everywhere instead of checking the attribute. I'll create a convenience wrapper around it.

Reply:
Yeah the name is pretty bad. IMO we should move to lowering *all* attributes early but it just hasn't happened. At the very least it shouldn't have "fn" in the name.